### ISSUE TITLE
store transmission down to 3476.20

### ISSUE THREAD

**Issue by londumas** [igmhub/LyaCoLoRe#71]
store transmission down to 3476.20, following the discussion on adding SiII(1260)

**Comment by jfarr03** [comment]
@londumas Can you explain where this number comes from again? Unless I'm mistaken, the wavelength cut required to observe SiII from 3600A on is given by:

3600*lambda_Lya/lambda_SiII = 3600*1215.67/1260 = 3473.34A

As such, would it be wise to choose a value just below this? Say 3470A, or 3450? Any value is fine in terms of LyaCoLoRe.

**Comment by andreufont** [comment]
Let's use 3470A,

**Comment by andreufont** [comment]
Sorry, I keep clicking on the wrong button on Github!

I was going to say that 1260A was probably just an approximation, the actual wavelength must have some decimal digits. 

I wouls just use 3470A and move on for now.

**Comment by jfarr03** [comment]
Sure - the value stored in LyaCoLoRe/py/metals.py is 1260.42, which gives 3472.19 as the minimum. I'll use 3470 for now but can easily change for any future versions of course.

**Comment by jfarr03** [comment]
One final clarification before I generate a new version: would we like to extend the wavelength range of the actual transmission files? Or just extend the skewers calculated within LyaCoLoRe and then use the same observed wavelength cut for the transmission files? 

As long as the longer skewers are calculated within LyaCoLoRe, the problem will be solved, and maintaining the current transmission file wavelength range (3550-6500A) avoids storing data that won't be used (i.e. the skewers from 3470-3550A). This seems like a sensible option to me as a result

However, the original request was to store the transmission itself to a lower wavelength, so I thought I'd check that the above solution is ok?

Also its worth noting that using 3470A means that there won't be any SiII absorption in the range 3550-3596A. If data below 3600A is thrown away anyway then it doesn't matter, of course, but its worth keeping in mind.

**Comment by andreufont** [comment]
Hi @jfarr03 - We should write to the transmission files everything to 3470A. 

Quickquasars will use the Lyman alpha absorption in the range (3470A,3600A) to compute the SiII(1260) absorption in the range (3600A,X). The output of quickquasars will indeed be limited to 3600A, but in order to do that we need a longer transmission file. 

By the way, @londumas - why did you choose 3600A? Isn't it true that the DESI spectrograph goes a little bit bluer? Is quickquasars generating flux only for lambda > 3600A? Or is this a data analysis cut in picca?

Helion will

**Comment by jfarr03** [comment]
Ah yes I forgot that the metals were being added via quickquasars for the moment and that needs the Lya absorption in the wavelength range. Once we use the metals within LyaCoLoRe storing the extra range won't be necessary any more as long as the skewers are kept sufficiently long internally.

**Comment by andreufont** [comment]
Exactly.

**Comment by londumas** [comment]
>By the way, @londumas - why did you choose 3600A? Isn't it true that the DESI spectrograph goes a little bit bluer? Is quickquasars generating flux only for lambda > 3600A? Or is this a data analysis cut in picca?

We can go lower than 3600 A for DESI, I used 3600A because that is what we use in eBOSS DR16 and in picca by default, and I did not want to ask for more than I need. But the more the better :) I have no strong opinion.

**Comment by jfarr03** [comment]
So for now I'll use 3470A as the cut (storing Lya absorption all the way to this wavelength), which will produce SiII absorption from 3597A onwards. In future, any value is possible, and the cut can be altered in relation to the transmission file range so that there is no sudden start/stop of metal absorption.

**Comment by andreufont** [comment]
@londumas - It might be good to print a warning flag in quickquasars if someone is trying to compute a metal absorption for a pixel corresponding to a metal wavelength that does not exist in the transmission file.

**Comment by londumas** [comment]
@andreufont, very good idea. I take care of that.

**Comment by londumas** [comment]
fixed in v7.0:
```
import fitsio
h = fitsio.FITS('/project/projectdirs/desi/mocks/lya_forest/london/v7.0/v7.0.0/0/0/transmission-16-0.fits')
h['WAVELENGTH'].read()
array([3470. , 3470.2, 3470.4, ..., 6499.4, 6499.6, 6499.8], dtype=float32)
```
